x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/index.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo } from 'react';
+import { ChoroplethMap } from './ChoroplethMap';
+import { ColorProgressionBar } from './ColorProgressionBar';
+import { useAvgDurationByCountry } from '../../../../../hooks/useAvgDurationByCountry';
+
+function getProgressionColor(
+  scale: number,
+  // default to euiColorPrimary (#006BB4)
+  color = { hue: 204, saturation: 100, lightness: { min: 20, max: 90 } }
+) {
+  const { min, max } = color.lightness;
+  const lightness = Math.round(min + (max - min) * (1 - scale));
+  return `hsl(${color.hue},${color.saturation}%,${lightness}%)`;
+}
+
+const RegionMapChartToolTip: React.SFC<{
+  geojsonProperties: { [name: string]: any };
+  data?: { key: string; value: number } & Partial<{ docCount: number }>;
```

Comment:
out of interest, why use `Partial<{}>` here and not `{key:string, value: number, docCount?: number}`?